WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewViewModel.swift

```diff
@@ -7,12 +7,29 @@ import protocol Storage.StorageManagerType
 /// View model for `ProductDetailPreviewView`
 ///
 final class ProductDetailPreviewViewModel: ObservableObject {
+    enum ProductDetailField: Equatable {
+        case name
+        case shortDescription
+        case description
+    }
+
+    struct NameSummaryDescOption {
+        let name: String
+        let shortDescription: String
+        let description: String
+    }
+
     typealias ImageState = EditableImageViewState
 
     @Published private(set) var imageState: ImageState
     @Published private(set) var isGeneratingDetails: Bool = false
     @Published private(set) var isSavingProduct: Bool = false
-    @Published private(set) var generatedProduct: Product?
+    @Published private(set) var generatedAIProduct: AIProduct?
+
+    @Published private(set) var selectedOptionIndex: Int = 0
+    @Published private var options: [NameSummaryDescOption] = [NameSummaryDescOption(name: "",
+                                                                                     shortDescription: "",
+                                                                                     description: "")]
```

Comment:
❓ I'm curious why we need an empty object for the default value instead of an empty array?

Reply:
Good point. An empty array is enough. Done in 580dd7448838c2334833c93a6a30e6883d4926c0